yaksh/views.py

```diff
@@ -1796,6 +1796,69 @@ def download_assignment_file(request, quiz_id, question_id=None, user_id=None):
 
 @login_required
 @email_verified
+def upload_users(request, course_id):
+    user = request.user
+    ci = RequestContext(request)
+    course = get_object_or_404(Course, pk=course_id)
+    context = {'course': course}
+
+    if not (course.is_teacher(user) or course.is_creator(user)):
+        msg = 'You do not have permissions to this course.'
+        return complete(request, reason=msg)
+    if request.method == 'POST':
+        if 'csv_file' not in request.FILES:
+            context['message'] = "Please upload a CSV file."
+            return my_render_to_response('yaksh/course_detail.html', context,
+                    context_instance=ci)
+        csv_file = request.FILES['csv_file']
+        if not is_csv(csv_file):
+            context['message'] = "The file uploaded is not a CSV file."
+            return my_render_to_response('yaksh/course_detail.html', context,
+                    context_instance=ci)
+        headers = {'firstname':'', 'lastname':'', 'email':''}
+        reader = csv.DictReader(csv_file.read().decode('utf-8').splitlines())
```

Comment:
This piques my interest!! Why does the decode thing not raise an error in python 3?